### PR TITLE
Auto-delete polls

### PR DESCRIPTION
**Pull request by Teccii**
- Made DSU automatically delete messages with polls (through raw event because polls are currently inaccessible through discord.js Message object).
- A command to enable/disable polls in text channels, voice channels, and threads.
- Polls are disabled by default in all channels.
- Threads inherit from their parent channel (text or forum) so polls can't be disabled in the thread if they are enabled in the parent channel, but can be enabled in the thread if they are disabled in the parent channel.
- Everyone with perm level Helper or above is immune and can post polls anywhere as they wish.
- Polls can only be enabled/disabled in a channel by someone with perm level Administrator or above.

### CODE REVIEW

File: src/events/raw.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    message.settings = s;
+  }
+
+  const pollsAllowed = message.settings.pollsAllowed;
+  const isThread = channel.type == ChannelType.PublicThread && channel.parent !== null;
+
+  if (!pollsAllowed.includes(channel.id) && (isThread && !pollsAllowed.includes(channel.parent.id))) {
+    await message.reply("You are not allowed to send polls in this channel.");
```
Comment (3vilN355) [inline src/events/raw.ts:61]: My only addendum to this pr is this; we should probably delete this message after a set timeout?
There's no "mute for spam" integrated in this and that's completely fine, however never deleting our message might perpetuate "spam" of this feature.

Comment (3vilN355) [reply src/events/raw.ts:61]: "this message" being our reply to the message (line 61), thought I was able to reference a single line in the github comment but apparently not

**Comment by 3vilN355** [comment]
Besides from my one comment, looks good to me

**Comment by Teccii** [comment]
ther

**Comment by Teccii** [comment]
I did test it. You can change the timeout's duration if you want to, I'm not terribly attached to it being 2 seconds instead of 5 seconds lol.

File: src/events/raw.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    !pollsAllowed.includes(channel.parent.id) &&
+    !pollsAllowed.includes(channel.id);
+
+  if (channelCondition || threadCondition) {
+    const msg = await message.reply("You are not allowed to send polls in this channel.");
+    await message.delete();
+
+    setTimeout(async () => {
+      await msg.delete();
+    }, 2000);
```
Comment (3vilN355) [inline src/events/raw.ts:71]:
```suggestion
    }, 5000);
```

**Pull request merged**